src/SignalR/server/SignalR/test/AddSignalRTests.cs

```diff
@@ -80,6 +80,20 @@ public void HubSpecificOptionsDoNotAffectGlobalHubOptions()
             Assert.Equal(1, serviceProvider.GetRequiredService<IOptions<HubOptions>>().Value.SupportedProtocols.Count);
             Assert.Equal(0, serviceProvider.GetRequiredService<IOptions<HubOptions<CustomHub>>>().Value.SupportedProtocols.Count);
         }
+
+        [Fact]
+        public void StreamBufferCapacityGetSet()
+        {
+            var serviceCollection = new ServiceCollection();
+
+            serviceCollection.AddSignalR().AddHubOptions<CustomHub>(options =>
+            {
+                options.StreamBufferCapacity = 42;
+            });
+
+            var serviceProvider = serviceCollection.BuildServiceProvider();
+            Assert.Equal(42, serviceProvider.GetRequiredService<IOptions<HubOptions<CustomHub>>>().Value.StreamBufferCapacity);
```

Comment:
Can we add a more end-to-end test that verifies the client observes the back pressure once the specified capacity is reached?

Reply:
Yeah, this still needs real tests. Should have included that in the description.

Reply:
@halter73 How would you expect to observe that backpressure?

Reply:
Once the bounded channel fills up, the DispatchMessagesAsync loop will wait until ChannelWriter.WriteAsync completes.

Eventually, if no one's reading HubMessages off of the transport's PipeReader, the transport induces backpressure on the client that prevents the client from writing new stream items, invocations or any other message until the backpressure is relieved.

Of course, the transport only sees the messages as bytes over a pipe, and if Kestrel tests have proven one thing: it's proven that it's hard to reliably test that backpressure is induced over an actual socket because there can be arbitrary amounts of buffering happening at different layers.

The real way to reliably test that the StreamBufferCapacity has an effect is to verify the [head-of-line blocking](https://en.wikipedia.org/wiki/Head-of-line_blocking) induced by the bounded channel. I would do this by:

1. Call a Hub method that accepts a stream but doesn't read it.
2. Send enough items from the client to completely fill the bounded Channel.
3. Try to invoke a second hub method.
4. Verify the second hub method doesn't get called.
5. Now read an item from the stream passed to the first hub method.
6. Verify the second hub method now does get called.

Long term, will probably want to use credit based flow control similar to HTTP/2 so clients know how many stream items they can send before inducing head-of-line blocking.